questions-3rd-ed.md

```diff
@@ -0,0 +1,221 @@
+
+- What are transaction inputs, and how are they constructed? What are unlocking scripts?
+
+--------------
+
+# Week 4
+
+## Chapter 7: Authorization and Authentication
+
+- Is the Turing incompleteness of Bitcoin's scripting language (SCRIPT) a feature or a deficiency?
+
+- What is the difference between a locking script (scriptPubKey) and an unlocking script (scriptSig)? What is the difference between a redeem script and a witness script?
+
+- Why are the locking and unlocking scripts executed separately?
+
+- Explain different types of timelocks which you can have and list the components of the transactions that can be used to set timelocks.
+
+- What are the different ways to create contracts in the script? Briefly highlight pros and cons of each.
+
+- Why is the public key directly included in a P2TR output whereas in other script types we always include a hash?
+
+- Why is the Merkle path length limited to 128?
+
+- How does Taproot upgrade increase security and privacy in Bitcoin?
+
+- Highlight the differences between Tapscript and native Bitcoin script?
+
+
+## Chapter 8: Digital Signature
+
+### Digital Signatures
+
+- What are sighashes, and what are their uses? How many types of sighashes are there?
+
+- What are digital signatures, where are they placed in a transaction, and what is their purpose?
+
+- Describe some real-life exploits that occurred due to poor randomness in digital signatures.
+
+- Besides signing transactions, what other uses could digital signatures have?
+
+
+--------------
+
+# Week 5
+
+## Chapter 9: Transaction Fess
+
+
+## Chapter 10: The Bitcoin Network
+
+- What is a peer-to-peer network? Give two examples of successful peer-to-peer networks that preceded Bitcoin.
+- What is a block finding race? How does compact block relay prevent it?
+- How does a new node discover other Bitcoin nodes on the network in order to participate?
+- How do light weight clients validate that a transaction exists? How is this different from the mechanism full archival nodes use to validate a transaction?
+- What are bloom filters? How do they work?
+- What are compact block filters? What data should be included in a block filter?
+- What are the privacy trade-offs in using a light weight client?
+- Explain in brief what is a Mempool and what is an Orphan Pool.
+--------------
+
+# Week 6
+
+## Chapter 11: The Blockchain
+
+- How is the Bitcoin blockchain analogous to layers in a geological formation?
+- Describe the structure of a block and what data does a block header contain?
+- What are the two identifiers of a Block? Discuss each in brief.
+- What is a Merkle Tree and what significance does it have in Bitcoin?
+- What is a design flaw in the Merkle Tree used in Bitcoin?
+- What is a Light Weight Client in context to Bitcoin? How do Light Weight Clients verify transactions associated to their addresses?
+- Explain mainnet and testnet in brief.
```

Comment:
A bit more explicite like

Whats the difference between mainnet and testnet? Can testnet coins have value?